WSL and Git Setup chapters (#168)
infinum/wordpress-handbook: Development environment setup/WSL Setup.md

There are two ways you can try to set up the dev environment on WSL, one is more foolproof, but takes a bit longer, the other is by just trying to set up [valet-wsl](https://github.com/valeryan/valet-wsl). It is faster, but we've had issues with it in the past.

## Installing the required tools

### WSL2

Make sure you have either Windows 10 or 11 installed.

For Windows 10, make sure you have:

- For x64 systems: version 1903 or higher, with build 18362 or higher
- For ARM64 systems: version 2004 or higher, with build 19041 or higher
- Builds lower than 18362 do not support WSL2, please update your Windows.

To check the version and build number, either open up a start menu and in the search bar, search for: `winver`, or open a run window (Windows key + R) and type `winver`.

Make sure you have virtualization enabled in your BIOS, otherwise WSL won't work as intended.

**Note**

In some cases, you might have to enable certain Windows features on. Most notably you should have `Virtual Machine Platform` and `Windows Subsystem for Linux` enabled in the Windows features list (they are located in the Control Panel -> Programs -> Turn Windows features on or off link).

---

> 

Reviewer: ```suggestion
In the case WSL is already installed you can check the list of available distros with `wsl --list --online` and you can run

```bash
wsl --install -d <DistroName>
```

to install a desired Linux distro.
```

Author: I wouldn't change the order of this, it's basically the same as in the official docs for installing WSL

Open the PowerShell or Windows Command Prompt **as an administrator** and type

```bash
wsl --install
```

This command will enable the required optional components, download the latest Linux kernel, set WSL 2 as your default, and install a Linux distribution for you (Ubuntu by default).

The first time you launch a newly installed Linux distribution, a console window will open, and you'll be asked to wait for files to de-compress and be stored on your machine. All future launches should take less than a second.

In the case there was some issue you can check the default WSL version with

```bash
wsl -l -v
```

> and if it's not set to 2, set it using the following command:

Reviewer: ```suggestion
Check if WSL version is set to 2, and if not set it using the following command:
```

Author: This is not a good change if you consider that this sentence is the part of the sentence above the code example. It would read oddly in that case I think

```bash
wsl --set-default-action 2
```

In the case WSL is already installed you can check the list of available distros with `wsl --list --online` and you can run

```bash

> wsl --instal -d <DistroName>

Reviewer 2: ```suggestion
wsl --install -d <DistroName>
```

Typo?

> In the case WSL is already installed you can check the list of available distros with `wsl --list --online` and you can run
> 
> ```bash
> wsl --install -d <DistroName>
> ```
> 
> to install a desired Linux distro.

Reviewer: ```suggestion
```
